src/Rings/mpolyquo-localizations.jl

```diff
@@ -2155,4 +2155,62 @@ function inverse(phi::MPolyQuoLocalizedRingHom)
   error("computation of inverse not implemented")
 end
 
+@doc raw"""
+    minimal_generating_set(I::MPolyLocalizedIdeal)
+
+Given an ideal `I` in the localization of a quotient of a  multivariate
+polynomial ring over a field at a point, return an array containing a
+minimal set of generators of `I`. If `I` is the zero ideal an empty list
+is returned.
+
+Note: This is only available for localizations at rational points. 
+"""
+function minimal_generating_set(
+    I::MPolyQuoLocalizedIdeal{<:MPolyQuoLocRing{<:Field, <:FieldElem,
+                                          <:MPolyRing, <:MPolyElem,
+                                          <:MPolyComplementOfKPointIdeal},
+                              <:Any,<:Any}
+  )
+  Q = base_ring(I)
+  L = localized_ring(Q)
+
+  J = pre_image_ideal(I)
+  return filter(!iszero, Q.(minimal_generating_set(J)))
+end
+
+@doc raw"""
+    small_generating_set(I::MPolyLocalizedIdeal)
+
+Given an ideal `I` in a localization of a multivariate polynomial ring
+over a field, return an array containing a set of generators of `I`,
+which is usually smaller than the original one.
 
+If `I` is the zero ideal an empty list is returned.
+
+If the localization is at a point, a minimal set of generators is returned.
+"""
+function small_generating_set(
+      I::MPolyQuoLocalizedIdeal{<:MPolyLocRing{<:Field, <:FieldElem,
+                                          <:MPolyRing, <:MPolyElem,
+                                          <:MPolyComplementOfKPointIdeal},
+                              <:Any,<:Any}
+  )
+  Q = base_ring(I)
+  L = localized_ring(Q)
+
+  J = ideal(L, lift.(gens(I))) + modulus(Q)
+  return filter(!iszero, Q.(small_generating_set(J)))
+end
+
+function small_generating_set(
```

Comment:
```suggestion
@attr function small_generating_set(
```
Maybe we should turn all these into attributes?

Reply:
Please add the type of the cached object after `@attr` to ensure type safety.

Reply:
done